packages/hooks/src/useHistoryTravel/index.ts

```diff
@@ -37,6 +37,24 @@ export default function useHistoryTravel<T>(initialValue?: T) {
   });
 
   const { present, past, future } = history;
+
+  const reset = useCallback(
+    (hasNewInitial: boolean = false, newInitialValue?: T) => {
+      const _initial = hasNewInitial
+        ? newInitialValue
+        : past.length
+        ? past.shift()
+        : present;
+
+      setHistory({
+        present: _initial,
+        future: [],
+        past: []
+      });
+    },
+    [history, setHistory]
+  );
+
```

Comment:
```suggestion

  const initialValueRef = useRef(initialValue);

  const reset = useCallback(
    () => {
      setHistory({
        present: initialValueRef.current,
        future: [],
        past: []
      });
    },
    [setHistory]
  );
```

reset 可以搞的简单一些，不需要修改初始值。

Reply:
without this, how can I implement a similar feature if the game board layout has changed?

```
const initialBoardData = useMemo(() => getBoardLayout(config), [config]);

const {
    value,
    setValue,
    backLength,
    forwardLength,
    back,
    forward,
    go,
    reset
} = useHistoryTravel({
    ...initialStateForTimeTravel,
    boardData: initialBoardData
});
```

Reply:
I have modified the suggested code~~

Reply:
@brickspert your suggestion is similar to my first attempt. But, I amended it after I realised from the tests that undefined is a valid initialValue. Reset should allow setting initialValue to undefined as well.

Reply:
reset is a bit troublesome to set two parameters  for one thing.

 or you can use like this

```js
const reset = (...params)=>{
   if(params.length > 0){
      ...
   }else{
      ...
   }
}
```

in this code , `reset()` is not equal to `reset(undefined)`.